plugins/arElasticSearchPlugin/lib/model/arElasticSearchActorPdo.class.php

```diff
@@ -352,6 +354,59 @@ public function serializeAltNames()
     return $serialized;
   }
 
+  public static function serializeObjectDirectRelationTypes($actorId, $relationData)
+  {
+    $relationTypeIds = [];
+
+    // Cycle through each relation an actor's involved in
+    foreach ($relationData as $relation)
+    {
+      $typeId = $relation['typeId'];
+
+      if ($relation['objectId'] == $actorId)
+      {
+        // Add type ID to array if it hasn't already been added
+        if (!in_array($typeId, $relationTypeIds))
+        {
+          $relationTypeIds[] = $typeId;
+        }
+      }
+      else
+      {
+        // If actor is the subject of the relation then look up the converse of the type
+        if (isset(self::$converseTermIds[$typeId]))
+        {
+          // Get cached type ID
+          $converseTermId = self::$converseTermIds[$typeId];
+        }
+        else
+        {
+          // Look up converse term, if any
+          $converseTermId = $typeId;
+
+          $relationTerm = QubitTerm::getById($typeId);
+          $converseTerm = $relationTerm->getConverseActorRelationTerm();
+
+          if ($converseTerm !== null)
+          {
+            $converseTermId = $converseTerm->id;
+          }
```

Comment:
It's great that you cached them but I think we should use ras SQL in here, even if they are a few it should be a pretty easy query.

Reply:
Good call.